src/UIHandler.cpp

```diff
@@ -0,0 +1,345 @@
+#include "UIHandler.h"
+
+UIWindow::UIWindow( int minSizeX, int minSizeY, Location location,
+                    bool drawBorder ) : m_panel( new UIPaddingPanel( drawBorder ) )
+{
+    m_minSize.x = minSizeX;
+    m_minSize.y = minSizeY;
+
+    m_thisLocation = location;
+
+    UpdateWindow();
+}
+
+int UIWindow::UpdateWindow()
+{
+    // TODO: Clear prev contents? Is that our job?
+
+    m_thisSize = m_panel->RequestedSize( Sizes::Prefered );
+
+    point minSize = m_panel->RequestedSize( Sizes::Minimum );
+
+    if( m_minSize.x > m_thisSize.x ) {
+        m_thisSize.x = m_minSize.x;
+    }
+    if( m_minSize.y > m_thisSize.y ) {
+        m_thisSize.y = m_minSize.y;
+    }
+
+    if( m_thisSize.x > TERMX ) {
+        m_thisSize.x = TERMX;
+    }
+    if( m_thisSize.y > TERMY ) {
+        m_thisSize.y = TERMY;
+    }
+
+    if( m_minSize.x > TERMX ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Not enought screen space...";
+    }
+    if( m_minSize.y > TERMY ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Not enought screen space...";
+    }
+
+    if( minSize.x > TERMX ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Not enought screen space...";
+    }
+    if( minSize.y > TERMY ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Not enought screen space...";
+    }
+
+    m_panel->SetSize( m_thisSize );
+
+    switch( m_thisLocation ) {
+        case Location::Centered:
+            m_offset.x = ( TERMX > m_thisSize.x ) ? ( TERMX - m_thisSize.x ) / 2 : 0;
+            m_offset.y = ( TERMY > m_thisSize.y ) ? ( TERMY - m_thisSize.y ) / 2 : 0;
+            break;
+
+        default:
+            DebugLog( D_ERROR, DC_ALL ) << "Please state the location...";
```

Comment:
Again it would be useful to know the invalid value.

Even better would be to just remove the `default` case and add a case for every enumeration entry. The compiler will tell you when you've missed one.